backend/test/features/activity-like.feature

```diff
@@ -1,42 +1,62 @@
-Feature: Like an object like an article or note
+Feature: Like an object e.g. an article or note
   As a user I want to like others posts
-  Also if I do not want to follow a previous followed user anymore,
-  I want to undo the follow.
+  Also if I decided to dislike a previous liked object,
+  I want to be able to do so.
 
   Background:
     Given our own server runs at "http://localhost:4123"
     And we have the following users in our database:
       | Slug           |
       | karl-heinz     |
-      | peter-lustiger |
-    And I send a POST request with the following activity to "/activitypub/users/bernd-das-brot/inbox":
+      | theodor        |
+    And I send a POST request with the following activity to "/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/karl-heinz/status/faslkasa7dasfzkjn2398hsfd",
+      "id": "http://localhost:4123/api/users/theodor/status/faslkasa7dasfzkjn2398hsfd",
       "type": "Create",
-      "actor": "http://localhost:4123/activitypub/users/karl-heinz",
+      "actor": "http://localhost:4123/users/theodor",
       "object": {
-          "id": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf",
-          "type": "Article",
-          "published": "2019-02-07T19:37:55.002Z",
-          "attributedTo": "http://localhost:4123/activitypub/users/karl-heinz",
-          "content": "Hi Max, how are you?",
-          "to": "https://www.w3.org/ns/activitystreams#Public"
+        "id": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf",
+        "type": "Article",
+        "published": "2019-02-07T19:37:55.002Z",
+        "attributedTo": "http://localhost:4123/api/users/theodor",
+        "content": "Test content for liking an object?",
+        "to": "https://www.w3.org/ns/activitystreams#Public"
       }
     }
     """
 
-  Scenario: Send a like of a person to an users inbox and make sure it's added to the likes collection
-    When I send a POST request with the following activity to "/activitypub/users/karl-heinz/inbox":
+  Scenario: Send a like of a person to a users inbox and make sure the object has been liked
+    When I send a POST request with the following activity to "/users/theodor/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/peter-lustiger/status/83J23549sda1k72fsa4567na42312455kad83",
+      "id": "http://localhost:4123/api/users/karl-heinz/status/83J23549sda1k72fsa4567na42312455kad83",
       "type": "Like",
-      "actor": "http://localhost:4123/activitypub/users/peter-lustiger",
-      "object": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf"
+      "actor": "http://localhost:4123/users/karl-heinz",
+      "object": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf"
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-05-06T23:17:34Z" title="Tuesday, May 7th 2019, 1:17:34 am +02:00">May 7, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,42 +1,62 @@
-Feature: Like an object like an article or note
+Feature: Like an object e.g. an article or note
   As a user I want to like others posts
-  Also if I do not want to follow a previous followed user anymore,
-  I want to undo the follow.
+  Also if I decided to dislike a previous liked object,
+  I want to be able to do so.
 
   Background:
     Given our own server runs at "http://localhost:4123"
     And we have the following users in our database:
       | Slug           |
       | karl-heinz     |
-      | peter-lustiger |
-    And I send a POST request with the following activity to "/activitypub/users/bernd-das-brot/inbox":
+      | theodor        |
+    And I send a POST request with the following activity to "/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/karl-heinz/status/faslkasa7dasfzkjn2398hsfd",
+      "id": "http://localhost:4123/api/users/theodor/status/faslkasa7dasfzkjn2398hsfd",
       "type": "Create",
-      "actor": "http://localhost:4123/activitypub/users/karl-heinz",
+      "actor": "http://localhost:4123/users/theodor",
       "object": {
-          "id": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf",
-          "type": "Article",
-          "published": "2019-02-07T19:37:55.002Z",
-          "attributedTo": "http://localhost:4123/activitypub/users/karl-heinz",
-          "content": "Hi Max, how are you?",
-          "to": "https://www.w3.org/ns/activitystreams#Public"
+        "id": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf",
+        "type": "Article",
+        "published": "2019-02-07T19:37:55.002Z",
+        "attributedTo": "http://localhost:4123/api/users/theodor",
+        "content": "Test content for liking an object?",
+        "to": "https://www.w3.org/ns/activitystreams#Public"
       }
     }
     """
 
-  Scenario: Send a like of a person to an users inbox and make sure it's added to the likes collection
-    When I send a POST request with the following activity to "/activitypub/users/karl-heinz/inbox":
+  Scenario: Send a like of a person to a users inbox and make sure the object has been liked
+    When I send a POST request with the following activity to "/users/theodor/inbox":
     """
     {
       "@context": "https://www.w3.org/ns/activitystreams",
-      "id": "http://localhost:4123/activitypub/users/peter-lustiger/status/83J23549sda1k72fsa4567na42312455kad83",
+      "id": "http://localhost:4123/api/users/karl-heinz/status/83J23549sda1k72fsa4567na42312455kad83",
       "type": "Like",
-      "actor": "http://localhost:4123/activitypub/users/peter-lustiger",
-      "object": "http://localhost:4123/activitypub/users/karl-heinz/status/dkasfljsdfaafg9843jknsdf"
+      "actor": "http://localhost:4123/users/karl-heinz",
+      "object": "http://localhost:4123/api/users/theodor/status/dkaaadsfljsdfaaaffg9843jknsdf"
```


@Mastercuber how come that sometimes you're routing to `http://localhost:4123/` and sometimes to `http://localhost:4123/api/`?

Reply:
> [<img alt="Mastercuber" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mastercuber) **Authored by [Mastercuber](https://github.com/Mastercuber)**
_<time datetime="2019-05-10T20:22:11Z" title="Friday, May 10th 2019, 10:22:11 pm +02:00">May 10, 2019</time>_

---


I've tried to fit the frontend url expectations (with `/api` prefixed), though the backend is tested. It's more for code understandability from the view of the client. The ports will also be changed to 3000.